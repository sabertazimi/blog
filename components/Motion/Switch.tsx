import type { MotionProps } from '@components/utils'
import { motion } from '@components/utils'
import type { AriaRole } from 'react'

interface Props {
  children?: React.ReactNode
  className?: string
  role?: AriaRole
  open?: boolean
  onClick?: () => void
  variants?: MotionProps['variants']
  transition?: MotionProps['transition']
}

const defaultVariants: MotionProps['variants'] = {
  open: {
    opacity: 1,
  },
  close: {
    opacity: 0,
  },
}

const defaultTransition: MotionProps['transition'] = {
  type: 'spring',
  duration: 0.2,
}

function Switch({
  children,
  className,
  role = 'switch',
  open = true,
  variants = defaultVariants,
  transition = defaultTransition,
  onClick,
  ...props
}: Props): JSX.Element {
  return (
    <motion.div
      className={className}
      role={role}
      initial="close"
      animate={open ? 'open' : 'close'}
      variants={variants}
      transition={transition}
      onClick={onClick}
      {...props}
    >
      {children}
    </motion.div>
  )
}

export default Switch
