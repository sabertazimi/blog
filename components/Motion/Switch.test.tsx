import { render } from '@utils'
import Switch from './Switch'

describe('switch', () => {
  it('should render correctly (snapshot)', () => {
    const { container } = render(<Switch />)

    expect(container).toMatchSnapshot()
  })
})
