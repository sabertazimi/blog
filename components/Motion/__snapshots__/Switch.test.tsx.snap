// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`switch should render correctly (snapshot) 1`] = `
<div>
  <div
    role="switch"
    style="opacity: 0;"
  />
</div>
`;
